New pages: HTMLInputElement checkedness (#35880)
mdn/content: files/en-us/web/api/htmlinputelement/checked/index.md

---
title: "HTMLInputElement: checked property"
short-title: checked
slug: Web/API/HTMLInputElement/checked
page-type: web-api-instance-property
browser-compat: api.HTMLInputElement.checked
---

{{ APIRef("HTML DOM") }}

> The **`checked`** property of the {{DOMxRef("HTMLInputElement")}} interface specifies the current checkedness of the element; whether the form control is checked or not. It reflects the {{htmlelement("input")}} element's [`checked`](/en-US/docs/Web/HTML/Element/input#checked) attribute.

Reviewer: ```suggestion
The **`checked`** property of the {{DOMxRef("HTMLInputElement")}} interface specifies the current checkedness of the element; that is, whether the form control is checked or not. It reflects the {{htmlelement("input")}} element's [`checked`](/en-US/docs/Web/HTML/Element/input#checked) attribute.
```

> The boolean `checked` attribute is valid for the `radio`([`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio)) and `checkbox` ([`<input type="checkbox">`](/en-US/docs/Web/HTML/Element/input/checkbox)) input types.

Reviewer: ```suggestion
The boolean `checked` attribute is valid for the `radio` ([`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio)) and `checkbox` ([`<input type="checkbox">`](/en-US/docs/Web/HTML/Element/input/checkbox)) input types.
```

> The presence of the boolean [`checked`](/en-US/docs/Web/HTML/Element/input#checked) attribute indicates the checkbox is checked by default. It does not indicate whether this checkbox is currently checked: if the checkbox's state is changed, this content attribute does not reflect the change; only the `HTMLInputElement`'s `checked` IDL attribute is updated.

Reviewer: ```suggestion
The presence of the HTML [`checked`](/en-US/docs/Web/HTML/Element/input#checked) attribute indicates the checkbox is checked by default. It does not indicate whether this checkbox is currently checked: if the checkbox's state is changed, this content attribute does not reflect the change; only the `HTMLInputElement`'s `checked` IDL property is updated. The `checked` attribute is reflected by the {{domxref("HTMLInputElement.defaultChecked", "defaultChecked")}} property.
```

> A boolean attribute indicating whether this checkbox is checked by default (when the page loads). It does not indicate whether this checkbox is currently checked: if the checkbox's state is changed, this content attribute does not reflect the change. (Only the HTMLInputElement's checked IDL attribute is updated.)

Reviewer: Then it does not "reflect" the attribute. An attribute is only "reflected" by a property if getting/setting the property is equivalent to getting/setting the attribute. Indeed, the `checked` attribute is reflected by the `defaultChecked` property, just like the `value` attribute is reflected by the `defaultValue` property.

Author: updated!
